src/config.rs

```diff
@@ -61,8 +61,68 @@ struct Root {
     clouds: Clouds,
 }
 
-fn find_config() -> Option<PathBuf> {
-    let current = Path::new("./clouds.yaml");
+/// Merge two nested serde_yaml::Mapping structs
+/// The values from src are merged into dest. Values in src override values in dest.
+fn merge_mappings(src: &serde_yaml::Mapping, dest: &mut serde_yaml::Mapping) {
+    for (src_key, src_value) in src.iter() {
+        if let Some(src_mapping) = src_value.as_mapping() {
+            if let Some(dest_value) = dest.get_mut(src_key) {
+                if let Some(dest_mapping) = dest_value.as_mapping_mut() {
+                    merge_mappings(src_mapping, dest_mapping);
+                    continue;
+                }
+            }
+        }
+        let _ = dest.insert(src_key.to_owned(), src_value.to_owned());
+    }
+}
+
+// Inject profiles from clouds-public.yaml into clouds.yaml and return it in a new Value
+fn inject_profiles(
+    clouds_public: &serde_yaml::Mapping,
+    clouds: &serde_yaml::Mapping,
```

Comment:
Similarly, I think both parameters can be taken by value.

Reply:
done!